File: Tellma.Api.Dto/PrintArguments.cs
using System.Collections.Generic;

namespace Tellma.Api.Dto
{
    public class PrintArguments
    {
        public string Culture { get; set; }

        public Dictionary<string, string> Custom { get; set; }
    }

    public class PrintEntitiesArguments<TKey> : PrintArguments
    {
        public string Filter { get; set; }
        public string OrderBy { get; set; }
        public int Top { get; set; }
        public int Skip { get; set; }
        public List<TKey> I { get; set; }
    }

    public class PrintEntityByIdArguments : PrintArguments
    {
    }

    public class PrintDynamicArguments : PrintArguments
    {
        public string Type { get; set; } // Summary/Details
        public string Select { get; set; }
        public string OrderBy { get; set; }
        public string Filter { get; set; }
        public string Having { get; set; }
        public int Top { get; set; }
        public int Skip { get; set; }
    }
}

File: Tellma.Api.Web/ClientApp/src/app/data/dto/print-arguments.ts
import { ReportArguments } from '../workspace.service';

export interface PrintArguments {
    culture?: string;
    version?: string;
}

export interface PrintEntitiesArguments extends PrintArguments {
    filter?: string;
    orderby?: string;
    top?: number;
    skip?: number;
    i?: (string | number)[];
}

// tslint:disable-next-line:no-empty-interface
export interface PrintEntityByIdArguments extends PrintArguments {
    // id?: string | number;
}
